src/unix/linux.c

```diff
@@ -2281,6 +2281,124 @@ uint64_t uv_get_available_memory(void) {
 }
 
 
+#define BUF_SIZE 1024
+
+static uv__cpu_constraint uv__get_cgroupv2_constrained_cpu(const char* cgroup) {
+  uv__cpu_constraint constraint = {0, 0, 0};
+  char path[256];
+  char buf[BUF_SIZE];
+  unsigned int weight;
+  int cgroup_size;
+  const char* cgroup_trimmed;
+
+  /* Trim ending \n by replacing it with a 0 */
+  cgroup_trimmed = cgroup + sizeof("0::/") - 1;      /* Skip the prefix "0::/" */
+  cgroup_size = (int)strcspn(cgroup_trimmed, "\n");  /* Find the first slash */
```

Comment:
This function assumes the buffer always starts with "0::/" but I don't see anything at the call site that guarantees that. You should check `if (strncmp(buf, "0::/", 4))` either here or (for consistency with existing code) there.

Reply:
Well in case of cgroup v2, `/sys/fs/cgroup/cgroup.controllers` [always starts with `0::/`, ](https://docs.kernel.org/admin-guide/cgroup-v2.html)
<img width="819" alt="image" src="https://github.com/libuv/libuv/assets/31201229/9bb61a92-89c2-43f5-b3e1-43d64607d63a">

Do I need that extra check ?

Reply:
Always validate that input is as expected. Someone may bind-mount another file over /sys/fs/cgroup/cgroup.controllers